src/main/java/homework2/RestarauntProcessor.java

```diff
@@ -0,0 +1,188 @@
+//            if(dish.getType() == DishType.BEEF){
+//                beef.add(dish);
+//            } else if (dish.getType() == DishType.CHICKEN) {
+//                chicken.add(dish);
+//            } else {
+//                vegetables.add(dish);
+//            }
+//        }
+//        groupByDishType.put(DishType.BEEF, beef);
+//        groupByDishType.put(DishType.CHICKEN, chicken);
+//        groupByDishType.put(DishType.VEGETABLES, vegetables);
+//
+//        for (Map.Entry<DishType, List<Dish>> entry : groupByDishType.entrySet()) {
+//            System.out.println(entry.getKey()+" : "+entry.getValue().toString());
+//        }
+//        return groupByDishType;
+
+        Map<DishType, List<Dish>> groupByDishType = restaraunt.getMenu().stream()
+                .collect(Collectors.groupingBy(Dish::getType, Collectors.toList()));
+
+        groupByDishType.entrySet().forEach(System.out::println);
+        return groupByDishType;
+    }
+
+    public static Map<DishType, List<String>> groupByBioName(Restaraunt restaraunt) {
+//        Map<DishType, List<String>> groupByBioName = new HashMap<>();
+//        List<String> beef = new ArrayList<>();
+//        List<String> chicken = new ArrayList<>();
+//        List<String> vegetables = new ArrayList<>();
+//        for (Dish dish : restaraunt.getMenu()) {
+//            if (dish.getType() == DishType.BEEF && dish.getIsBio()) {
+//                beef.add(dish.getName());
+//            } else if (dish.getType() == DishType.CHICKEN && dish.getIsBio()) {
+//                chicken.add(dish.getName());
+//            } else if (dish.getIsBio()) {
+//                vegetables.add(dish.getName());
+//            }
+//        }
+//        groupByBioName.put(DishType.BEEF, beef);
+//        groupByBioName.put(DishType.CHICKEN, chicken);
+//        groupByBioName.put(DishType.VEGETABLES, vegetables);
+//        for (Map.Entry<DishType, List<String>> entry : groupByBioName.entrySet()) {
+//            System.out.println(entry.getKey() + " : " + entry.getValue().toString());
+//        }
+//        return groupByBioName;
+
+        Map<DishType, List<String>> groupByBioName = restaraunt.getMenu().stream()
+                .filter(Dish::getIsBio)
+                .collect(Collectors.groupingBy(Dish::getType, Collectors.mapping(Dish::getName, Collectors.toList())));
+
+        groupByBioName.entrySet().forEach(System.out::println);
+        return groupByBioName;
+    }
+
+    private static Comparator<Dish> getCompCalories()
+    {
+        return (d1, d2) -> d2.getCalories().compareTo(d1.getCalories());
+    }
+    private static Comparator<Dish> getCompType()
+    {
+        return Comparator.comparing(Dish::getType);
+    }
+
+    private static Comparator<Dish> getCompName()
+    {
+        return Comparator.comparing(Dish::getName);
+    }
+
+    private static double calcAverageCalories(List<Integer> dish) {
+        if (dish == null || dish.isEmpty())
```

Comment:
it is better to rename `dish` -> `dishes`
Also, you should not do null checks for collection types.  As well as not passing `null` for collection types. Usually, the empty collection fits better.